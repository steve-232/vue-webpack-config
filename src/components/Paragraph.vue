<template>
  <p :style="{ fontSize, color }">
    <slot />
  </p>
</template>

<script>
export default {
  props: {
    fontSize: {
      type: String,
      default: '16px',
    },
    color: {
      type: String,
      default: '#000',
    },
  },
};
</script>
