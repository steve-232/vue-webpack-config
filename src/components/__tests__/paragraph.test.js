import { mount } from '@vue/test-utils';
import Paragraph from '../Paragraph';

describe('Paragraph.vue', () => {
  it('render custom text & add custom css style & snapshot', () => {
    const wrapper = mount(Paragraph, {
      slots: {
        default: 'Lorem ipsum'
      },
      propsData: {
        fontSize: '20px',
        color: '#09f'
      },
    });

    // custom text
    expect(wrapper.text()).toBe('Lorem ipsum');

    // custom style
    expect(wrapper.element.style._values).toHaveProperty('font-size', '20px');
    expect(wrapper.element.style._values).toHaveProperty('color', 'rgb(0, 153, 255)');

    // snapshot
    expect(wrapper).toMatchSnapshot();
  })
});