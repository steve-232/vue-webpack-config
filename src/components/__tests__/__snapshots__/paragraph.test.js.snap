// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph.vue render custom text & add custom css style & snapshot 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
